test/http-request.ts

```diff
@@ -0,0 +1,332 @@
+
+describe('get trace and span from http-request', () => {
+  const FAKE_CONTEXT = {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    extract: (headerWrapper: {}) => {},
+    generate: () => {},
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    inject: (headerWrapper: {}, spanContext: {}) => {},
+  };
+
+  const fakeContext = Object.assign({}, FAKE_CONTEXT);
+
+  const {getOrInjectTraceParent} = proxyquire('../src/http-request', {
+    '@opencensus/propagation-stackdriver': fakeContext,
+  });
+
+  describe('getTraceContext', () => {
+    it('should return an empty trace string when extraction fails and inject is false', () => {
+      const req = {
+        method: 'GET',
+      } as http.IncomingMessage;
+      const context = getTraceContext(req, 'myProj');
+      assert.strictEqual(context.trace, '');
+      assert.strictEqual(context.spanId, undefined);
+      assert.strictEqual(context.traceSampled, undefined);
+    });
+
+    it('should return a formatted Google Cloud trace context first', () => {
+      const req = {
+        headers: {['x-cloud-trace-context']: '1/2;o=1'},
+      } as unknown as http.IncomingMessage;
+      const projectId = 'myProj';
+      const context = getTraceContext(req, projectId);
+      assert.strictEqual(context.trace, `projects/${projectId}/traces/1`);
+      assert.strictEqual(context.spanId, '2');
+      assert.strictEqual(context.traceSampled, true);
+    });
+
+    it('should return a formatted W3C trace context next', () => {
+      const req = {headers: {}} as http.IncomingMessage;
+      // This should generate a span and trace if not available.
+      const context = getTraceContext(req, 'myProj', true);
+      assert(context.trace.includes('projects/myProj/traces/'));
+      assert(context.spanId!.length > 0);
+      assert.strictEqual(context.traceSampled, undefined);
+    });
+  });
+
+  describe('makeHeaderWrapper', () => {
+    const HEADER_NAME = 'Content-Type';
+    const HEADER_VALUE = 'application/🎂';
+
+    it('should correctly get request headers', () => {
+      const req = {headers: {[HEADER_NAME]: HEADER_VALUE}};
+      const wrapper = makeHeaderWrapper(req as unknown as http.IncomingMessage);
+      assert.strictEqual(wrapper!.getHeader(HEADER_NAME), HEADER_VALUE);
+    });
+
+    it('should correctly set request headers', () => {
+      const req = {headers: {} as http.IncomingHttpHeaders};
+      const wrapper = makeHeaderWrapper(req as unknown as http.IncomingMessage);
+      wrapper!.setHeader(HEADER_NAME, HEADER_VALUE);
+      assert.strictEqual(req.headers[HEADER_NAME], HEADER_VALUE);
+    });
+
+    it('should return null if header property is not in http request', () => {
+      const req = {
+        method: 'GET',
+      } as http.IncomingMessage;
+      const wrapper = makeHeaderWrapper(req as unknown as http.IncomingMessage);
+      assert.strictEqual(wrapper, null);
+    });
+  });
+
+  describe('getCloudTraceContext', () => {
```

Comment:
Perhaps, worth being more descriptive here

Reply:
the describe block should reflect the name of the function we're testing. I'll add some descriptive comments within the block